phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -153,6 +344,15 @@ public void join(MutationState newMutationState) {
         if (this == newMutationState) { // Doesn't make sense
             return;
         }
+        // TODO: what if new and old have txContext as that's really an error
+        // Really it's an error if newMutation txContext is not null
+        if (txContext != null) {
```

Comment:
@JamesRTaylor  should I add a check if the newMutation  txContext is not null and throw an IllegalStateException?